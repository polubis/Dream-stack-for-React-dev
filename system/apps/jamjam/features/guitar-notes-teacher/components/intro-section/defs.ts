interface IntroSectionProps {
  header: string;
  description: string;
  action: string;
  onConfirm: () => void;
}

export type { IntroSectionProps };
