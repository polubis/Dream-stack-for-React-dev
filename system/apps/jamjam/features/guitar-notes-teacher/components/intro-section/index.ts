export * from './defs';
export * from './intro-section';
