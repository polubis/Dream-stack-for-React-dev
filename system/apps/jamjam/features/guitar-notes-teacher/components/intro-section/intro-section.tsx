import { Box, Button, Font } from '@system/figa-ui';
import styled from 'styled-components';

import type { IntroSectionProps } from './defs';

const Container = styled.div`
  max-width: 400px;
  margin: auto;
`;

const IntroSection = ({
  header,
  description,
  action,
  onConfirm,
}: IntroSectionProps) => {
  return (
    <Container>
      <Box spacing={[150, 500]} padding={[350, 350, 350, 350]}>
        <Font variant="h6">{header}</Font>
        <Font variant="b1">{description}</Font>
        <Button onClick={onConfirm}>{action}</Button>
      </Box>
    </Container>
  );
};

export { IntroSection };
