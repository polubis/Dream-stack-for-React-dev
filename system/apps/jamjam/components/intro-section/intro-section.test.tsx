import { fireEvent, render, screen } from '@testing-library/react';

import { IntroSection } from './intro-section';
import type { IntroSectionProps } from './defs';

describe('User is able to user intro section when', () => {
  const introSectionFixture = (props: Partial<IntroSectionProps> = {}) => {
    const result = render(
      <IntroSection
        header="header"
        description="description"
        action="action"
        onConfirm={jest.fn()}
        {...props}
      />
    );

    return result;
  };

  it('head, description and title of interactive element are displayed', () => {
    introSectionFixture();

    screen.getByText(/header/);
    screen.getByText(/description/);
    screen.getByText(/action/);
  });

  it('when confirmation can be clicked', () => {
    const onConfirmSpy = jest.fn();

    introSectionFixture({ onConfirm: onConfirmSpy });

    fireEvent.click(screen.getByText(/action/));

    expect(onConfirmSpy).toHaveBeenCalledTimes(1);
  });

  it('[FRAGILE] classes are assigned and box component is used', () => {
    const { asFragment } = introSectionFixture();

    expect(asFragment()).toMatchSnapshot();
  });
});
