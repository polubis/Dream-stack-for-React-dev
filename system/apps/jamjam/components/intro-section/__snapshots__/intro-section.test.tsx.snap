// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`User is able to user intro section when [FRAGILE] classes are assigned and box component is used 1`] = `
<DocumentFragment>
  <div
    class="sc-dkrFOg kfxkmd"
  >
    <div
      class="box column empty"
      style="padding: 28px 28px 28px 28px;"
    >
      <div
        class="box-item-wrapper"
        style="margin-bottom: 12px;"
      >
        <h6
          class="font font-h6"
        >
          header
        </h6>
      </div>
      <div
        class="box-item-wrapper"
        style="margin-bottom: 40px;"
      >
        <span
          class="font font-b1"
        >
          description
        </span>
      </div>
      <div
        class="box-item-wrapper"
      >
        <button
          class="button button-size-3 button-rectangle button-filled button-primary"
        >
          action
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
